Intermittent CI failures

CI Smoke tests on main appear to be failing intermittently

June 4th: 🔴 - https://github.com/logos-messaging/libchat/actions/runs/26988915201
June 5th: 🟢 - https://github.com/logos-messaging/libchat/actions/runs/27047992214
June 6th: 🔴 - https://github.com/logos-messaging/libchat/actions/runs/27078698886

Logs show:

```
...
building '/nix/store/s7m237p1wp0m79sa1ldygkqpj39wdfjd-nim-stint-470b789.drv'...
building '/nix/store/s6f30wrgjkx7p9fda0gd7x4ah38r4kg9-nim-taskpools-9e8ccc7.drv'...
error: Cannot build '/nix/store/lwhy2446pv500gnnl30m5sjxlz3cg5zk-nim-snappy-00bfcef.drv'.
       Reason: builder failed with exit code 123.
       Output paths:
         /nix/store/77knvp73x3w2ks6hccc19j8i4c9fb1hn-nim-snappy-00bfcef
       Last 25 log lines:
       > remote: Total 280 (delta 82), reused 73 (delta 27), pack-reused 0 (from 0)
       > Receiving objects: 100% (280/280), 989.16 KiB | 3.01 MiB/s, done.
       > Resolving deltas: 100% (82/82), done.
       > remote: Total 0 (delta 0), reused 0 (delta 0), pack-reused 0 (from 0)
       > remote: Enumerating objects: 253, done.
       > remote: Counting objects: 100% (253/253), done.
       > remote: Compressing objects: 100% (121/121), done.
       > remote: Total 131 (delta 112), reused 13 (delta 6), pack-reused 0 (from 0)
       > Receiving objects: 100% (131/131), 44.68 KiB | 1.35 MiB/s, done.
       > Resolving deltas: 100% (112/112), completed with 106 local objects.
       > From https://github.com/google/benchmark
       >  * branch            b20cea674170b2ba45da0dfaf03953cdea473d0d -> FETCH_HEAD
       > Submodule path 'tests/snappycpp/third_party/benchmark': checked out 'b20cea674170b2ba45da0dfaf03953cdea473d0d'
       > remote: Total 0 (delta 0), reused 0 (delta 0), pack-reused 0 (from 0)
       > remote: Enumerating objects: 356, done.
       > remote: Counting objects: 100% (356/356), done.
       > remote: Compressing objects: 100% (164/164), done.
       > Receiving objects: 100% (184/184), 103.70 KiB | 1.62 MiB/s, done.
       > remote: Total 184 (delta 158), reused 29 (delta 16), pack-reused 0 (from 0)
       > Resolving deltas: 100% (158/158), completed with 142 local objects.
       > From https://github.com/google/googletest
       >  * branch            b796f7d44681514f58a683a3a71ff17c94edb0c1 -> FETCH_HEAD
       > Submodule path 'tests/snappycpp/third_party/googletest': checked out 'b796f7d44681514f58a683a3a71ff17c94edb0c1'
       > removing `.git'...
       > rm: cannot remove '/nix/store/77knvp73x3w2ks6hccc19j8i4c9fb1hn-nim-snappy-00bfcef/.git': Directory not empty
       For full logs, run:
         nix log /nix/store/lwhy2446pv500gnnl30m5sjxlz3cg5zk-nim-snappy-00bfcef.drv
building '/nix/store/88jgabci6fix9zv4krv9kglixkq1b44h-nim-testutils-6ce5e5e.drv'...
error: Cannot build '/nix/store/jakp05ya7z63njj1p0w2i28hvvkimsmv-liblogosdelivery-dev.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/5rsm6gyw3cr8k54pr6kzmah735rra03d-liblogosdelivery-dev
error: Build failed due to failed dependency
Error: Process completed with exit code 1.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -74,7 +74,7 @@ jobs:
         run: make valgrind
         working-directory: crates/client-ffi/examples/message-exchange
       - name: Build logos-delivery
-        # TEST: build through a patched nixpkgs (kaichaosun/nixpkgs fix-gitfetch),
+        # Build through a patched nixpkgs (kaichaosun/nixpkgs fix-gitfetch),
         # whose nix-prefetch-git disables git background auto-maintenance so the
         # nim-zlib submodule fetch no longer races `.git` removal. The override
         # flows into logos-delivery via its `nixpkgs.follows`. The bumped cache
@@ -110,6 +110,6 @@ jobs:
         with:
           primary-key: nix-${{ runner.os }}-fixtest-${{ hashFiles('flake.nix', 'flake.lock') }}
           restore-prefixes-first-match: nix-${{ runner.os }}-
-      # TEST: same patched-nixpkgs override; the default package pulls in
+      # Same patched-nixpkgs override; the default package pulls in
       # logos-delivery-lib, so it exercises the same nim-zlib fetch.
       - run: nix build --override-input nixpkgs github:kaichaosun/nixpkgs/fix-gitfetch --print-build-logs

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -6,7 +6,7 @@ on:
   pull_request:
     branches: [main]
   schedule:
-    - cron: '0 0 * * *'  # Every night at midnight UTC
+    - cron: "0 0 * * *" # Every night at midnight UTC
 
 env:
   CARGO_TERM_COLOR: always
@@ -58,7 +58,7 @@ jobs:
             experimental-features = nix-command flakes
       - uses: nix-community/cache-nix-action@v6
         with:
-          primary-key: nix-${{ runner.os }}-${{ hashFiles('flake.nix', 'flake.lock') }}
+          primary-key: nix-${{ runner.os }}-fixtest-${{ hashFiles('flake.nix', 'flake.lock') }}
           restore-prefixes-first-match: nix-${{ runner.os }}-
       - name: Install valgrind
         if: runner.os == 'Linux'
@@ -74,7 +74,15 @@ jobs:
         run: make valgrind
         working-directory: crates/client-ffi/examples/message-exchange
       - name: Build logos-delivery
-        run: nix build .#logos-delivery
+        # TEST: build through a patched nixpkgs (kaichaosun/nixpkgs fix-gitfetch),
+        # whose nix-prefetch-git disables git background auto-maintenance so the
+        # nim-zlib submodule fetch no longer races `.git` removal. The override
+        # flows into logos-delivery via its `nixpkgs.follows`. The bumped cache
+        # key (`-fixtest-`) forces a cold store, so this genuinely re-fetches
+        # nim-zlib instead of reusing a cached output. No retry on purpose: a
+        # single build must now succeed deterministically. Re-run the job a few
+        # times for more samples.
+        run: nix build .#logos-delivery --override-input nixpkgs github:kaichaosun/nixpkgs/fix-gitfetch --print-build-logs
       # Build and run chat-cli through the dev shell so it links against the
       # same Nix glibc as the prebuilt liblogosdelivery.so. A plain `cargo
       # build` uses the runner's system glibc, which is older than Nix's and
@@ -100,6 +108,8 @@ jobs:
             experimental-features = nix-command flakes
       - uses: nix-community/cache-nix-action@v6
         with:
-          primary-key: nix-${{ runner.os }}-${{ hashFiles('flake.nix', 'flake.lock') }}
+          primary-key: nix-${{ runner.os }}-fixtest-${{ hashFiles('flake.nix', 'flake.lock') }}
           restore-prefixes-first-match: nix-${{ runner.os }}-
-      - run: nix build --print-build-logs
+      # TEST: same patched-nixpkgs override; the default package pulls in
+      # logos-delivery-lib, so it exercises the same nim-zlib fetch.
+      - run: nix build --override-input nixpkgs github:kaichaosun/nixpkgs/fix-gitfetch --print-build-logs

